src/files-card-body.tsx

```diff
@@ -407,6 +420,66 @@ const getFileType = (file: FolderFileInfo) => {
     }
 };
 
+const FilePermissions = ({ file } : {
+    file: FolderFileInfo,
+}) => {
+    function permissionShortStr(mode: number) {
+        const specialBits = (mode >> 9) & 0b111;
+        const permsStr = [];
+        for (let i = 2; i >= 0; i--) {
+            const offset = i * 3;
+            let shortStr = "";
+            shortStr += (mode & (0b1 << (offset + 2))) ? "r" : "-";
+            shortStr += (mode & (0b1 << (offset + 1))) ? "w" : "-";
+
+            if (mode & (1 << offset)) {
+                if (specialBits & (0b1 << i)) {
+                    shortStr += (i === 0) ? "t" : "s";
+                } else {
+                    shortStr += "x";
+                }
+            } else {
+                if (specialBits & (0b1 << i)) {
+                    shortStr += (i === 0) ? "T" : "S";
+                } else {
+                    shortStr += "-";
+                }
+            }
+
+            permsStr.push(shortStr);
+        }
+
+        return permsStr;
+    }
+
+    const { mode } = file;
+    if (mode === undefined) {
```

Comment:
hmm that is just `file.mode`? Feels a bit unnatural :)

Reply:
at least I changed it to `const mode = file.mode`. Using `file.mode` makes typescript unhappy for some reason (as you could see in the screenshot I sent on matrix)

Reply:
Can't figure this out either..